src/btc-faucet.ts

```diff
@@ -45,47 +45,100 @@ export function getRpcClient(): RPCClient {
   return client;
 }
 
+interface TxOutUnspent {
+  amount: number;
+  desc: string;
+  height: number;
+  scriptPubKey: string;
+  txid: string;
+  vout: number;
+}
+
 interface TxOutSet {
   bestblock: string;
   height: number;
   success: boolean;
   total_amount: number;
   txouts: number;
-  unspents: {
-    amount: number;
-    desc: string;
-    height: number;
-    scriptPubKey: string;
-    txid: string;
-    vout: number;
-  }[];
+  unspents: TxOutUnspent[];
 }
 
 // Replace with client.estimatesmartfee() for testnet/mainnet
 const REGTEST_FEE_RATE = 2000;
 
 const MIN_TX_CONFIRMATIONS = 100;
 
-export async function makeBTCFaucetPayment(
+function isValidBtcAddress(network: btc.Network, address: string): boolean {
+  try {
+    btc.address.toOutputScript(address, network);
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
+export async function getBtcBalance(network: btc.Network, address: string) {
+  if (!isValidBtcAddress(network, address)) {
+    throw new Error(`Invalid BTC regtest address: ${address}`);
+  }
+  const client = getRpcClient();
+
+  const txOutSet: TxOutSet = await client.scantxoutset({
+    action: 'start',
+    scanobjects: [`addr(${address})`],
+  });
```

Comment:
> Should not you be checking txOutSet.success and ensure that the scan was completed, and if it's false, poll with action = "status" as long as you don't reach 100%?

Yes this should be done, eventually. However, it is not a cause of bugs for now. The symptom of this bug would be the error below stating not enough balance from utxos. 

Again, wallet functions like `importaddress` could be used which would make this entire file a few lines of code that call `importaddress`, `listunspent` and `sendtoaddress`. The problem is that most wallet functions should not be publicly exposed, because they are trivial security (or at the very least simple DDoS) attack vectors -- much more so than the set of RPC calls currently used. 

If we are okay with this situation, then it's trivial to refactor the faucet into a few lines of wallet RPC calls. Just note that anyone interested could write a trivial script that at the very least disrupts the neon.

Reply:
The RPC call has been fixed to check for success/progress. The discussion around wallet RPC functions still needs resolved, I think. @lgalabru LMK